R/import_processed_data_for_PeCorA.R

```diff
@@ -0,0 +1,63 @@
+#' @title import_processed_data_for_PeCorA
+#' @description reads peptide.txt MaxQuant output example file inside the package and process them for PeCorA
+#' @param x peptide.txt MaxQuant output file
+#' @author Maria Dermit <maria.dermit@qmul.ac.uk>
+#' @return dataframe ready for PeCorA analysis
+#' @details output files containing columns: Peptide.Modified.Sequence", Protein, variable,value,Condition,BioReplicate
+#' @examples
+#' \dontrun{
+#' if(interactive()){
+#' t<-import_processed_data_for_PeCorA("peptides.txt")
+#'  }
+#' }
+#' @rdname import_processed_data_for_PeCorA
+#' @importFrom reshape melt
+#'@export
+import_processed_data_for_PeCorA <- function(x){
+  mypath <- system.file(
+    "extdata",
+    x,
+    package = "PeCorA"
+  )
+
+  p <- read.delim(
+    mypath,
+    stringsAsFactors = FALSE,
+  )
+  grep(colnames(p), pattern="LFQ")
+  colnames(p)[1:50]
+  idcol = c("Sequence", "Leading.razor.protein")
+
+  pmelt <-reshape::melt(p,
+               id.vars=idcol,
+               measure.vars=grep(colnames(p), pattern="LFQ"))
+  head(pmelt)
+
+  levels(as.factor(pmelt$variable))
+
+  pmelt$Condition = rep(0, nrow(pmelt))
+  pmelt$Condition[grep(pmelt$variable, pattern="_COVID19")]<-"COVID"
+  head(pmelt)
+  pmelt$Condition[grep(pmelt$variable, pattern="NON.COVID19")]<-"NONCO"
+  pmelt$Condition[grep(pmelt$variable, pattern="control")]<-"CNTRL"
```

Comment:
Need to update this so the function can account for user-specific groups in their maxquant data.

Reply:
Changed in function `import_processed_data_for_PeCorA_LFQ`

i.e.
t2 <- import_processed_data_for_PeCorA_LFQ( LFQfile = "peptides.txt",
condition1="CONTROL",
condition2="_COVID",
condition3="NON.COVID")